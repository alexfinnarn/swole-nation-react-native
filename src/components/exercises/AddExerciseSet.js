import React, {useState} from 'react';
import {Text, TextInput, View, Picker, Button} from "react-native";
import shortId from 'shortid';
import {styles} from "../Styles";

export default function AddExerciseSet({item = {key: "0", reps: 0, weight: 100}, exercise, toAdd, updater}) {
  const [reps, setReps] = useState(item.reps.toString());
  const [weight, setWeight] = useState(item.weight.toString());

  function updatePicker(value, type) {
    if (type === 'reps') {
      setReps(value);
      updater({reps: value, weight: weight, key: item.key}, exercise, shortId.generate(), 'UPDATE_SET');
    } else {
      setWeight(value);
      updater({reps: reps, weight: value, key: item.key}, exercise, shortId.generate(), 'UPDATE_SET');
    }
  }

  return (
    <View style={{flex: 1, flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', padding: 5, borderWidth: 1, borderColor: '#d6d7da', borderRadius: 4}}>
      <Text>Reps</Text>
      <Picker
        selectedValue={reps}
        style={{height: 60, width: 120}}
        onValueChange={(itemValue) => updatePicker(itemValue, 'reps')}>
        {[...Array(12)].map((e, index) => <Picker.Item key={index.toString()} label={index.toString()} value={index.toString()}/>)}
      </Picker>
      <Text>Weight</Text>
      <Picker
        selectedValue={weight}
        style={{height: 60, width: 120}}
        onValueChange={(itemValue) => updatePicker(itemValue, 'weight')}>
        {[...Array(75)].map((e, index) => <Picker.Item key={(index * 5).toString()} label={(index * 5).toString()} value={(index * 5).toString()}/>)}
      </Picker>
      {toAdd
        ? <Button onPress={() => {
          updater({reps: reps, weight: weight}, exercise, shortId.generate(), 'ADD_SET')
        }} title="+"/>
        : <Button onPress={() => {
          updater({key: item.key, reps: reps, weight: weight}, exercise, shortId.generate(), 'REMOVE_SET')
        }} title="X"/>
      }
    </View>

  );
}

