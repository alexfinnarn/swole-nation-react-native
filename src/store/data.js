import addFive from '../transformers/addFive'

const data = {
  workouts: {
    DQkECwYLCQQ: {
      name: "Stronglifts A",
      key: "DQkECwYLCQQ",
      description: "Squats, Bench Press, Barbell Row",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Bench Press Warmup",
        "Bench Press",
        "Barbell Row Warmup",
        "Barbell Row"
      ]
    },
    DAwJBQEMAA0: {
      name: "Stronglifts B",
      key: "DAwJBQEMAA0",
      description: "Squats, Overhead Press, Deadlifts",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Overhead Press Warmup",
        "Overhead Press",
        "Deadlifts Warmup",
        "Deadlifts"
      ]
    }
  },
  exercises: {
    BwEBDwoFBgQ: {
      name: "Squats Warmup",
      key: "BwEBDwoFBgQ",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    },
    Bw0ECwAKBws: {
      name: "Squats",
      key: "Bw0ECwAKBws",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        }
      ]
    },
    BwwBDAIJDgw: {
      name: "Bench Press Warmup",
      key: "BwwBDAIJDgw",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 3,
          "weight": 135.0
        }
      ]
    },
    DgcBCA8BCQo: {
      name: "Bench Press",
      key: "DgcBCA8BCQo",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        }
      ]
    },
    DwcACw4PCgI: {
      name: "Shoulder Press Warmup",
      key: "DwcACw4PCgI",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 3,
          "weight": 75.0
        }
      ]
    },
    CAMBAwsFBwQ: {
      name: "Shoulder Press",
      key: "CAMBAwsFBwQ",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        }
      ]
    },
    BwYGAw0NDwo: {
      name: "Barbell Row Warmup",
      key: "BwYGAw0NDwo",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 95.0
        }
      ]
    },
    BwQKAwYKAgA: {
      name: "Barbell Row",
      key: "BwQKAwYKAgA",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 115.0
        },
        {
          "reps": 5,
          "weight": 115.0
        }
      ]
    },
    AQcNAQkBCwo: {
      name: "Deadlifts Warmup",
      key: "AQcNAQkBCwo",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 5,
          "weight": 185.0
        },
        {
          "reps": 5,
          "weight": 205.0
        }
      ]
    },
    DAYCDAQKAw4: {
      name: "Deadlifts",
      key: "DAYCDAQKAw4",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 225.0
        }
      ]
    }
  },
  sessions: {
    BwgFDQ8CCg8: {
      name: "new ...date",
      key: "BwgFDQ8CCg8",
      workoutKeys: [],
      exercises: []
    }
  },
  transformers: {
    none: {
      key: 'none',
      label: "N/A",
      callback: null
    },
    addFive: {
      key: 'addFive',
      label: "Exercises (+5 lbs)",
      callback: addFive
    }
  },
  activeWorkoutKey: "",
  activeExerciseKey: "",
  activeTransformerKey: "none",
};

export default data;
