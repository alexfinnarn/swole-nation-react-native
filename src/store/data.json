{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
