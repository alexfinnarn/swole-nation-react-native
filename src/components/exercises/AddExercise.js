import React, {useState, useEffect} from 'react';
import {Text, View, TouchableOpacity, Keyboard, Switch, TextInput, Button, FlatList} from "react-native";
import Autocomplete from 'react-native-autocomplete-input';
import shortId from 'shortid';
import {styles} from "../Styles";
import AddExerciseSet from "./AddExerciseSet";

export default function AddExercise({exercises, thing, handleUpdate, addExercise, navigation, theExercise}) {
  console.log(theExercise);

  const [query, setQuery] = useState(theExercise.name);
  const [hideResults, setHideResults] = useState(false);
  const [exercise, setExercise] = useState(theExercise);
  const [switchValue, setSwitchValue] = useState(true);
  const exerciseList = findExercise(query);
  const comp = (a, b) => a.toLowerCase().trim() === b.toLowerCase().trim();

  function findExercise(query) {
    if (query === '') {
      return [];
    }

    const regex = new RegExp(`${query.trim()}`, 'i');
    return exercises.filter(exercise => exercise.name.search(regex) >= 0);
  }

  function updateAutocompleteInput(name) {
    setQuery(name);
    setExercise(exercises.find(exercise => exercise.name.toLowerCase().trim() === name.toLowerCase().trim()));
    setHideResults(true);
    Keyboard.dismiss();
  }

  function changeTheText(text) {
    setQuery(text);
    setHideResults(false);
  }

  return (
    <View style={{flex: 1, flexDirection: 'column'}}>
      <View style={{flex: 1, flexDirection: 'row'}}>
        {switchValue
          ? <Autocomplete
            autoCapitalize="none"
            autoCorrect={false}
            containerStyle={styles.autocompleteContainer}
            hideResults={hideResults}
            data={exerciseList.length === 1 && comp(query, exerciseList[0].name) ? [] : exerciseList}
            defaultValue={query}
            onChangeText={text => changeTheText(text)}
            placeholder="Enter Exercise"
            renderItem={({item: {name}}) => (
              <TouchableOpacity style={{paddingTop: 10, paddingBottom: 10}} key={shortId.generate()} onPress={() => updateAutocompleteInput(name)}>
                <Text style={styles.itemText}>{name}</Text>
              </TouchableOpacity>
            )}
          />
          : <TextInput
            style={[styles.editText, styles.smallTextInputFont, {flex: 4}]}
            placeholder="Enter Exercise"
            onChangeText={(text) => setQuery(text)}
            value={query}
          />
        }
        <Switch style={{flex: 1}} onValueChange={(value) => setSwitchValue(value)} value={switchValue}/>
      </View>
      <View style={{flex: 6}}>
        {exercise.name && <Text>Sets</Text>}
        <FlatList
          data={exercise.sets}
          extraData={thing}
          keyExtractor={(item, index) => item.key}
          renderItem={({item}) => <AddExerciseSet updater={handleUpdate} exercise={exercise} item={item}/>}
        />
      </View>
      <View style={{flex: 1}}>
        <Text>Add Exercise:</Text>
        <AddExerciseSet updater={handleUpdate} exercise={exercise} toAdd={true}/>
      </View>
      <View style={{flex: 1, justifyContent: "flex-end"}}>
        <Button onPress={() => {
          addExercise(exercise);
          navigation.goBack();
        }} title="Add" />
      </View>
    </View>
  );
}
