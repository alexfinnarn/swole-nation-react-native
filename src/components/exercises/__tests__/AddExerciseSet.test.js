import React from 'react';
import AddExerciseSet from '../AddExerciseSet';
import {render, fireEvent} from 'react-native-testing-library';
import data from '../../../store/data';
import {Picker} from "react-native";
import AddExercise from "../AddExercise";

let renderer, instance = {};
const updater = jest.fn();


describe('<AddExerciseSet />', () => {
  it('Renders correctly', () => {
    renderer = render(<AddExerciseSet updater={updater} toAdd={true}/>);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('renders default values with toAdd', () => {
    renderer = render(<AddExerciseSet updater={updater} toAdd={true}/>);
    instance = renderer.getByTestId('add-exercise-set-root');

    expect(renderer.getByTestId('reps-picker').props.selectedValue).toBe('0');
    expect(renderer.getByTestId('weight-picker').props.selectedValue).toBe('100');
    expect(renderer.getByText('+')).toBeTruthy();
  });

  it('renders passed in item with toAdd false', () => {
    const item = {item: {reps: 5, weight: 145}, index: 1};
    renderer = render(<AddExerciseSet item={item} updater={updater} toAdd={false}/>);
    instance = renderer.getByTestId('add-exercise-set-root');

    expect(renderer.getByTestId('reps-picker').props.selectedValue).toBe('5');
    expect(renderer.getByTestId('weight-picker').props.selectedValue).toBe('145');

    fireEvent(renderer.getByText('X'), 'press');
    expect(updater).toHaveBeenCalledTimes(1);
    expect(updater).toHaveBeenLastCalledWith({reps: '5', weight: '145'}, 1, 'REMOVE_SET');
  });

  it('updates set weight and reps and saves', () => {
    renderer = render(<AddExerciseSet updater={updater} toAdd={true}/>);
    instance = renderer.getByTestId('add-exercise-set-root');

    const repsPicker = renderer.getByTestId('reps-picker');
    const weightPicker = renderer.getByTestId('weight-picker');

    fireEvent(repsPicker, 'valueChange', '4');
    expect(repsPicker.props.selectedValue).toBe('4');

    fireEvent(weightPicker, 'valueChange', '160');
    expect(weightPicker.props.selectedValue).toBe('160');

    fireEvent(renderer.getByText('+'), 'press');

    // Should not be called the previous two times on changing reps and weight numbers.
    expect(updater).toHaveBeenCalledTimes(2);
    expect(updater).toHaveBeenLastCalledWith({reps: '4', weight: '160'}, 0, 'ADD_SET');
  });
});

