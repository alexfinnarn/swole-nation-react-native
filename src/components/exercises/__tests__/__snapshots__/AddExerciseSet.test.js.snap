// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddExerciseSet /> Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 5,
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 4,
        "marginTop": 4,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "space-between",
      },
    ]
  }
  testID="add-exercise-set-root"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "paddingLeft": 5,
        }
      }
    >
      Reps
    </Text>
    <View
      style={
        Object {
          "flex": 2,
          "height": 60,
          "width": 120,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "0",
              "textColor": undefined,
              "value": "0",
            },
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
            Object {
              "label": "4",
              "textColor": undefined,
              "value": "4",
            },
            Object {
              "label": "5",
              "textColor": undefined,
              "value": "5",
            },
            Object {
              "label": "6",
              "textColor": undefined,
              "value": "6",
            },
            Object {
              "label": "7",
              "textColor": undefined,
              "value": "7",
            },
            Object {
              "label": "8",
              "textColor": undefined,
              "value": "8",
            },
            Object {
              "label": "9",
              "textColor": undefined,
              "value": "9",
            },
            Object {
              "label": "10",
              "textColor": undefined,
              "value": "10",
            },
            Object {
              "label": "11",
              "textColor": undefined,
              "value": "11",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="reps-picker"
      />
    </View>
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Weight
    </Text>
    <View
      style={
        Object {
          "flex": 2,
          "height": 60,
          "width": 120,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "0",
              "textColor": undefined,
              "value": "0",
            },
            Object {
              "label": "5",
              "textColor": undefined,
              "value": "5",
            },
            Object {
              "label": "10",
              "textColor": undefined,
              "value": "10",
            },
            Object {
              "label": "15",
              "textColor": undefined,
              "value": "15",
            },
            Object {
              "label": "20",
              "textColor": undefined,
              "value": "20",
            },
            Object {
              "label": "25",
              "textColor": undefined,
              "value": "25",
            },
            Object {
              "label": "30",
              "textColor": undefined,
              "value": "30",
            },
            Object {
              "label": "35",
              "textColor": undefined,
              "value": "35",
            },
            Object {
              "label": "40",
              "textColor": undefined,
              "value": "40",
            },
            Object {
              "label": "45",
              "textColor": undefined,
              "value": "45",
            },
            Object {
              "label": "50",
              "textColor": undefined,
              "value": "50",
            },
            Object {
              "label": "55",
              "textColor": undefined,
              "value": "55",
            },
            Object {
              "label": "60",
              "textColor": undefined,
              "value": "60",
            },
            Object {
              "label": "65",
              "textColor": undefined,
              "value": "65",
            },
            Object {
              "label": "70",
              "textColor": undefined,
              "value": "70",
            },
            Object {
              "label": "75",
              "textColor": undefined,
              "value": "75",
            },
            Object {
              "label": "80",
              "textColor": undefined,
              "value": "80",
            },
            Object {
              "label": "85",
              "textColor": undefined,
              "value": "85",
            },
            Object {
              "label": "90",
              "textColor": undefined,
              "value": "90",
            },
            Object {
              "label": "95",
              "textColor": undefined,
              "value": "95",
            },
            Object {
              "label": "100",
              "textColor": undefined,
              "value": "100",
            },
            Object {
              "label": "105",
              "textColor": undefined,
              "value": "105",
            },
            Object {
              "label": "110",
              "textColor": undefined,
              "value": "110",
            },
            Object {
              "label": "115",
              "textColor": undefined,
              "value": "115",
            },
            Object {
              "label": "120",
              "textColor": undefined,
              "value": "120",
            },
            Object {
              "label": "125",
              "textColor": undefined,
              "value": "125",
            },
            Object {
              "label": "130",
              "textColor": undefined,
              "value": "130",
            },
            Object {
              "label": "135",
              "textColor": undefined,
              "value": "135",
            },
            Object {
              "label": "140",
              "textColor": undefined,
              "value": "140",
            },
            Object {
              "label": "145",
              "textColor": undefined,
              "value": "145",
            },
            Object {
              "label": "150",
              "textColor": undefined,
              "value": "150",
            },
            Object {
              "label": "155",
              "textColor": undefined,
              "value": "155",
            },
            Object {
              "label": "160",
              "textColor": undefined,
              "value": "160",
            },
            Object {
              "label": "165",
              "textColor": undefined,
              "value": "165",
            },
            Object {
              "label": "170",
              "textColor": undefined,
              "value": "170",
            },
            Object {
              "label": "175",
              "textColor": undefined,
              "value": "175",
            },
            Object {
              "label": "180",
              "textColor": undefined,
              "value": "180",
            },
            Object {
              "label": "185",
              "textColor": undefined,
              "value": "185",
            },
            Object {
              "label": "190",
              "textColor": undefined,
              "value": "190",
            },
            Object {
              "label": "195",
              "textColor": undefined,
              "value": "195",
            },
            Object {
              "label": "200",
              "textColor": undefined,
              "value": "200",
            },
            Object {
              "label": "205",
              "textColor": undefined,
              "value": "205",
            },
            Object {
              "label": "210",
              "textColor": undefined,
              "value": "210",
            },
            Object {
              "label": "215",
              "textColor": undefined,
              "value": "215",
            },
            Object {
              "label": "220",
              "textColor": undefined,
              "value": "220",
            },
            Object {
              "label": "225",
              "textColor": undefined,
              "value": "225",
            },
            Object {
              "label": "230",
              "textColor": undefined,
              "value": "230",
            },
            Object {
              "label": "235",
              "textColor": undefined,
              "value": "235",
            },
            Object {
              "label": "240",
              "textColor": undefined,
              "value": "240",
            },
            Object {
              "label": "245",
              "textColor": undefined,
              "value": "245",
            },
            Object {
              "label": "250",
              "textColor": undefined,
              "value": "250",
            },
            Object {
              "label": "255",
              "textColor": undefined,
              "value": "255",
            },
            Object {
              "label": "260",
              "textColor": undefined,
              "value": "260",
            },
            Object {
              "label": "265",
              "textColor": undefined,
              "value": "265",
            },
            Object {
              "label": "270",
              "textColor": undefined,
              "value": "270",
            },
            Object {
              "label": "275",
              "textColor": undefined,
              "value": "275",
            },
            Object {
              "label": "280",
              "textColor": undefined,
              "value": "280",
            },
            Object {
              "label": "285",
              "textColor": undefined,
              "value": "285",
            },
            Object {
              "label": "290",
              "textColor": undefined,
              "value": "290",
            },
            Object {
              "label": "295",
              "textColor": undefined,
              "value": "295",
            },
            Object {
              "label": "300",
              "textColor": undefined,
              "value": "300",
            },
            Object {
              "label": "305",
              "textColor": undefined,
              "value": "305",
            },
            Object {
              "label": "310",
              "textColor": undefined,
              "value": "310",
            },
            Object {
              "label": "315",
              "textColor": undefined,
              "value": "315",
            },
            Object {
              "label": "320",
              "textColor": undefined,
              "value": "320",
            },
            Object {
              "label": "325",
              "textColor": undefined,
              "value": "325",
            },
            Object {
              "label": "330",
              "textColor": undefined,
              "value": "330",
            },
            Object {
              "label": "335",
              "textColor": undefined,
              "value": "335",
            },
            Object {
              "label": "340",
              "textColor": undefined,
              "value": "340",
            },
            Object {
              "label": "345",
              "textColor": undefined,
              "value": "345",
            },
            Object {
              "label": "350",
              "textColor": undefined,
              "value": "350",
            },
            Object {
              "label": "355",
              "textColor": undefined,
              "value": "355",
            },
            Object {
              "label": "360",
              "textColor": undefined,
              "value": "360",
            },
            Object {
              "label": "365",
              "textColor": undefined,
              "value": "365",
            },
            Object {
              "label": "370",
              "textColor": undefined,
              "value": "370",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={20}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="weight-picker"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
